pkg/workload/schemachange/tracing.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package schemachange
+
+import (
+	"context"
+
+	"github.com/cockroachdb/errors"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// EndSpan is a helper for ending a span and annotating it with error
+// information, if relevant. Usage:
+//
+//	func mytracedfunc(ctx context.Context) (retErr err) {
+//		ctx, span := tracer.Start(ctx, "mytracedfunc")
+//		defer func() { EndSpan(span, retErr) }()
+//	}
+func EndSpan(span trace.Span, err error) {
+	if err == nil {
+		span.SetStatus(codes.Ok, "")
+	} else {
+		span.RecordError(err)
+		span.SetStatus(codes.Error, err.Error())
+
+		// If the error is a context cancellation, add an attribute so we can
+		// ignore the "errors".
+		// TODO(chrisseto): Maybe we should just use codes.Unset and record a
+		// "cancellation" event?
+		if errors.Is(err, context.Canceled) {
```

Comment:
> Maybe we should just use codes.Unset and record a
	"cancellation" event?

I would agree with this comment - I feel like more context is better whenever I am trying to debug things; ignoring the error might lead to more confusion from a beginners perspective

Maybe we can just leave this as is and if things do end up getting confusing, we can do this TODO?

Reply:
Went ahead and fixed this as I was running into issues later on 😅